examples/AI-surrogate-mode-hard-magnets.ipynb

```diff
@@ -0,0 +1,384 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "6f55f51b",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "# Hard magnet tutorial - with AI-based property predictions\n",
+    "\n",
+    "## Introduction\n",
+    "\n",
+    "- This notebook demonstrates how to use the `mammos-ai` package to predict hard magnet properties using pre-trained machine learning models.\n",
+    "- This is the same workflow as the [hard magnet tutorial](https://mammos-project.github.io/mammos/demonstrator/hard-magnet-tutorial.html) but uses AI models to predict hysteresis properties instead of micromagnetic simulations.\n",
+    "\n",
+    "\n",
+    "Requirements:\n",
+    "- Software: `mammos`\n",
+    "- Basic understanding of [hard magnet tutorial](https://mammos-project.github.io/mammos/demonstrator/hard-magnet-tutorial.html)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "2ee01e2d",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "%config InlineBackend.figure_format = \"retina\"\n",
+    "\n",
+    "\n",
+    "import mammos_ai\n",
+    "import mammos_analysis\n",
+    "import mammos_dft\n",
+    "import mammos_entity as me\n",
+    "import mammos_spindynamics\n",
+    "import mammos_units as u\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "6637a1cb",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Allow convenient conversions between A/m and T\n",
+    "u.set_enabled_equivalencies(u.magnetic_flux_field());"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "eff2974e",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "## Load data from DFT and spindynamics databases"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "6de4d152",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "material = \"Co2Fe2H4\"\n",
+    "\n",
+    "results_dft = mammos_dft.db.get_micromagnetic_properties(material)\n",
+    "results_spindynamics = mammos_spindynamics.db.get_spontaneous_magnetization(material)"
```

Comment:
Question beyond this notebook: are there any plans to also replace this line with an AI model?